src/prg_quantumdynamics_mod.F90

```diff
@@ -322,112 +202,114 @@ end subroutine prg_commutate_bml
   !! \param dt the timestep for integration
   !! \param hbar the Dirac constant (generally taken to be 1 in simulation units)
   !! \param rho_old the density matrix at previous time-step
-  !! \param rho_cur the density matrix at current time-step
-  !! \param rho_new the density matrix at next time-step
+  !! \param rho_bml the density matrix at current time-step
+  !! \param aux_bml the temp matrix used for value storage during computation
+  !! \param matrix_type the type of BML matrix
+  !! \param thresh the threshold for the BML matrix
   !!
-  subroutine prg_lvni_bml(H, sinv_bml, dt, hbar, rho_old, rho_cur,aux_bml,matrix_type)
-    implicit none
+  subroutine prg_lvni_bml(H, sinv_bml, dt, hbar, rho_old, rho_bml,aux_bml,matrix_type,M,thresh)
+    integer                             :: norbs, M
     real(dp)                            :: hbar, dt
     type(bml_matrix_t)                  :: H, sinv_bml
-    type(bml_matrix_t)                  :: rho_cur, rho_old, aux_bml
+    type(bml_matrix_t)                  :: rho_bml, rho_old, aux_bml,aux_1
     complex(dp)                         :: dR
+    real(dp), intent(in)                :: thresh
     character(len=*), intent(in)        :: matrix_type
-
+    
+    norbs=bml_get_N(rho_bml)
     dR = 2.0_dp*dt*cmplx(0.0_dp,-1.0_dp/hbar)
-    call prg_commutate_bml(H,rho_cur,aux_bml)
-    call bml_scale_cmplx(dR,aux_bml)
-    call bml_add(aux_bml,rho_old,1.0d0,1.0d0)
-    call bml_copy(rho_cur,rho_old)
-    call bml_copy(aux_bml,rho_cur)
-
+    call bml_multiply(sinv_bml,H,aux_bml,1.0_dp,0.0_dp,thresh)
+    call bml_zero_matrix(matrix_type,BML_ELEMENT_COMPLEX,dp,norbs,M,aux_1)
+    call bml_multiply(aux_bml,rho_bml,aux_1,1.0_dp,0.0_dp,thresh)
+    call bml_multiply(rho_bml,H,aux_bml,1.0_dp,0.0_dp,thresh)
+    call bml_multiply(aux_bml,sinv_bml,aux_1,-1.0_dp,1.0d0)
+    call bml_scale(dR,aux_1,aux_bml)
```

Comment:
The compilation is failing here in this line. Ah, ok, sorry, that's my fault. I need to update the test environment with the new version of the bml. I'll re-run this test after I am done with that.